collections/forms/src/validators/integer.js

```diff
@@ -5,8 +5,13 @@ const invalidIntegerMessage = i18n.t(
     'Please provide a round number without decimals'
 )
 
+const DECIMAL_SEPARATOR_PATTERN = /(\.|,)/
+
 const integer = (value) =>
-    isEmpty(value) || (isNumeric(value) && isInteger(toNumber(value)))
+    isEmpty(value) ||
+    (!DECIMAL_SEPARATOR_PATTERN.test(value) &&
+        isNumeric(value) &&
+        isInteger(toNumber(value)))
```

Comment:
I think instead of checking for decimal deparators, we can adjust the `isInteger` helper:

```js
const INTEGER_PATTERN = /^[1-9][0-9]*$/
export const isInteger = (value) => (
    Number.isSafeInteger(value) &&
    INTEGER_PATTERN.test(value.toString())
)
```

Reply:
Putting the logic in the `isInteger` helper doesn't quite work unless we refactor even more because in the validator, we're checking whether `isInteger(toNumber(value))` is true so the `toNumber(value)` bit would convert something like `4.0` to `4` first, making it be accepted.

As for the RegExp itself, I agree it's easier to rationalise positive booleans (has numbers) than negative booleans (does not have decimal)  .. but it's just that the regular expression in this case has to be more complicated than the one you have, we have to cater for negative integers (so allow `-` at the start) and also allow `0`. It felt it's easier to express what we don't want (the decimal separator) rather than what we want in this case.

Reply:
> Putting the logic in the isInteger helper doesn't quite work unless we refactor even more because in the validator, we're checking whether isInteger(toNumber(value)) is true so the toNumber(value) bit would convert something like 4.0 to 4 first, making it be accepted.

You're absolutely correct! I didn't see that.

> It felt it's easier to express what we don't want (the decimal separator) rather than what we want in this case.

I suppose your current implementation is good enough. We can change it to `/^-?[0-9]+$/` if there's ever any issue with it